### PR TITLE
Feature/160 replier reconnect

### PR DESCRIPTION
**Pull request by adenh93**

### CODE REVIEW

File: client/src/keep_alive/helpers.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        err.kind(),
+        io::ErrorKind::ConnectionReset | io::ErrorKind::NotConnected
+    )
+}
+
+pub fn is_bind_error(code: u32) -> bool {
+    matches!(code, REPLIER_ALREADY_BOUND)
+}
+
+pub fn is_recoverable_error(err: &SeliumError) -> bool {
```
Comment (petehayes102) [inline client/src/keep_alive/helpers.rs:16]: Would it be cleaner to just use a single match statement here?

File: client/src/keep_alive/reqrep.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+use super::helpers::is_recoverable_error;
+use crate::request_reply::{Replier, Requestor};
+use crate::traits::KeepAliveStream;
+use futures::Future;
+use selium_std::errors::QuicError;
+use selium_std::errors::Result;
+use selium_std::traits::codec::{MessageDecoder, MessageEncoder};
+use std::fmt::Debug;
+
+pub struct KeepAliveReqRep<T> {
```
Comment (petehayes102) [inline client/src/keep_alive/reqrep.rs:11]: 1. Should this also be named `KeepAlive` for consistency?
2. You're missing the `doc(hidden)` annotation.

File: server/src/server.rs [hunk 3]
```diff
@@ -166,27 +168,35 @@ async fn handle_stream(
...
         {
+            use selium_protocol::error_codes::REPLIER_ALREADY_BOUND;
+
             // Note this can only occur if someone circumvents the client lib
             if !topic.is_valid() {
-                stream
-                    .send(Frame::Error(b"Invalid topic name".to_vec().into()))
-                    .await?;
+                let payload = ErrorPayload {
+                    code: REPLIER_ALREADY_BOUND,
```
Comment (petehayes102) [inline server/src/server.rs:196]: Is this the error code you want? Shouldn't it be `INVALID_TOPIC_NAME`?

**Review by petehayes102 — CHANGES_REQUESTED** [review]
I can't believe you churned this out so quick! I really like the new abstractions around error handling / keep alive. We are crying out for a formal protocol spec, but all in good time I guess.

File: server/src/server.rs (hunk 3 above)
Comment (adenh93) [reply server/src/server.rs:196]: Yes, yes it should 🤦‍♂️

File: client/src/keep_alive/reqrep.rs (hunk 2 above)
Comment (adenh93) [reply client/src/keep_alive/reqrep.rs:11]: Yep, good point! We can avoid collisions via the namespace 👌

File: client/src/keep_alive/helpers.rs (hunk 1 above)
Comment (adenh93) [reply client/src/keep_alive/helpers.rs:16]: Yep! I was originally going to add more codes to match on, and forgot to clean up after that wasn't the case 👍 Fixed!

File: server/src/server.rs (hunk 3 above)
Comment (adenh93) [reply server/src/server.rs]: Fixed!

File: client/src/keep_alive/reqrep.rs (hunk 2 above)
Comment (adenh93) [reply client/src/keep_alive/reqrep.rs]: Added both changes 👍

**Review by petehayes102 — APPROVED** [review]

**Pull request merged**